### ISSUE TITLE
Consistent QT include pathing

### ISSUE THREAD

**Issue by hytel** [stachenov/quazip#98]
In, for example, quazip_qt_compat.h, there are includes like <QtCore/QTextCodec> and <QtCore/QSaveFile>.  These are great and fully qualify which Qt classes that need to be included.  Other includes like <Qt> and <QtGlobal> are also present which really should be <QtCore/Qt> and <QtCore/QtGlobal> I think.  I really do not wish to add an absolute Qt include subpath to my large project because the wrong thing may be pulled given that I am using many large 3rd party libraries.  The two part include path is great because you really shouldn't have any conflict in a large project due to the uniqueness of the Qt paths.  Any way we can standardize them to be fully qualified throughout?

**Comment by stachenov** [comment]
Sure thing, consistency is always a good thing. Just send a PR and I'll merge it right away.

**Comment by hytel** [comment]
I haven’t actually done a pull on your project, just a clone.  Instead, for now, I’ve temporarily added QtCore as an include directory.  I haven’t looked at the Quazip project as a whole, but my guess is that the QtCore directory is being directly included which is why <Qt> and <QtGlobal> work for you out-of-the-box.  I’m only using quazip as an external 3rdParty so I got missing header errors all over the place once I included “quazip.h”.  I looked back at the cmake resulting solution that I compiled and I see “C:\Qt\Qt5.8.0\5.8\msvc2015_64\include “ but I also see an extra “C:\Qt\Qt5.8.0\5.8\msvc2015_64\include\QtCore” in there which somehow gets generated during your cmake process.  This is lucky, but I would imagine most Qt projects don’t add each component as hardcoded include directories to search.  Interestingly enough many of your includes are fully pathed. If you just remove that all Qt component includes, you’ll see each location where you didn’t specifically mention the component in the include path.  Is it possible you could try that?  Would probably take you about 10 minutes to correct them. The Quazip project and testing harness isn’t that large,

Hillel Steinberg
Senior Software Engineer
OVERWATCH
An Operating Unit of Textron Systems
21660 Ridgetop Circle, Suite 110
Sterling, VA  20166
Office: 703 437-7651 x2441
Fax: 703 437-0039
hsteinberg@overwatch.textron.com<mailto:hsteinberg@overwatch.textron.com>
www.overwatch.com<http://www.overwatch.com/>

 “WARNING: Documents that can be viewed, printed or retrieved from this E-Mail may contain technical data whose export is restricted by the Arms Export Control Act (Title 22, U.S.C., Sec 2751, et seq,) or the Export Administration Act of 1979, as amended, Title 50, U.S.C., App. 2401 et seq. and which may not be exported, released or disclosed to non-U.S. persons (i.e. persons who are not U.S. citizens or lawful permanent residents [“green card” holders]) inside or outside the United States, without first obtaining an export license.  Violations of these export laws are subject to severe civil, criminal and administrative penalties.”

**Comment by stachenov** [comment]
The thing is, CMake adds those automatically. Probably with these lines in Qt5CoreConfig.cmake:
```
    set(_Qt5Core_OWN_INCLUDE_DIRS "${_qt5Core_install_prefix}/include/" "${_qt5Core_install_prefix}/include/QtCore") 
    set(Qt5Core_INCLUDE_DIRS ${_Qt5Core_OWN_INCLUDE_DIRS}) 
```
I don't really see any way of disabling this, short of manually editing these files.

Since your project already have this problem, all you really need to do is to fork QuaZip, clone the fork into your project, create a branch, fix whatever needs to be fixed, and push that branch back into your GitHub account. After that, issuing a PR is just a matter of clicking a single button.

But I see that you're probably new to GitHub, and the process can be confusing at first, so I'll try to do it myself just by looking over the sources and headers, fixing whatever I can see. I'll let you know when I'm done.

**Comment by hytel** [comment]
Thanks!  I have been using GitHub for a bit, but mostly to clone things.  But, as I said, you have an easy solution:


  1.  Generate the normal Quazip Solution using the cmake process.
  2.  Open Up the project (Visual Studio?) and remove the extra includes beyond the top-level one.
  3.  Compile again to expose the non-fully-qualified includes.

   I do need to know the process that you describe below.  It’s on my plate, but I don’t want to make a mistake that would affect others right now.  I would have thought others would have been affected by this, but I guess I’m one of the few who isn’t consuming quazip via the cmake process,

Hillel Steinberg
Senior Software Engineer
OVERWATCH
An Operating Unit of Textron Systems
21660 Ridgetop Circle, Suite 110
Sterling, VA  20166
Office: 703 437-7651 x2441
Fax: 703 437-0039
hsteinberg@overwatch.textron.com<mailto:hsteinberg@overwatch.textron.com>
www.overwatch.com<http://www.overwatch.com/>

 “WARNING: Documents that can be viewed, printed or retrieved from this E-Mail may contain technical data whose export is restricted by the Arms Export Control Act (Title 22, U.S.C., Sec 2751, et seq,) or the Export Administration Act of 1979, as amended, Title 50, U.S.C., App. 2401 et seq. and which may not be exported, released or disclosed to non-U.S. persons (i.e. persons who are not U.S. citizens or lawful permanent residents [“green card” holders]) inside or outside the United States, without first obtaining an export license.  Violations of these export laws are subject to severe civil, criminal and administrative penalties.”

**Comment by stachenov** [comment]
Messing with a build system doesn't sound like an easy way to me... I use CMake to both generate makefiles and to build the whole thing, both from command line and from Qt Creator.

Anyway, I've just converted everything to the `<QtModule/QClass>` scheme. Hope I haven't missed anything. Try it out and let me know, I'll delete the feature branch then.

**Comment by hytel** [comment]
Sergei ,

I downloaded the latest and saw all the great header file changes.  I then removed the extra forced header path I had added previously and recompiled.  Looks good, no issues compiling your quazip solution or my project with the header updates and only the top-level Qt incude in dir path in my application.  Thanks a bunch for your prompt response!

Hillel Steinberg
Senior Software Engineer
OVERWATCH
An Operating Unit of Textron Systems
21660 Ridgetop Circle, Suite 110
Sterling, VA  20166
Office: 703 437-7651 x2441
Fax: 703 437-0039
hsteinberg@overwatch.textron.com<mailto:hsteinberg@overwatch.textron.com>
www.overwatch.com<http://www.overwatch.com/>

 “WARNING: Documents that can be viewed, printed or retrieved from this E-Mail may contain technical data whose export is restricted by the Arms Export Control Act (Title 22, U.S.C., Sec 2751, et seq,) or the Export Administration Act of 1979, as amended, Title 50, U.S.C., App. 2401 et seq. and which may not be exported, released or disclosed to non-U.S. persons (i.e. persons who are not U.S. citizens or lawful permanent residents [“green card” holders]) inside or outside the United States, without first obtaining an export license.  Violations of these export laws are subject to severe civil, criminal and administrative penalties.”
